src/caffe/layers/data_layer.cpp

```diff
@@ -1,3 +1,4 @@
+#ifdef USE_OPENCV
```

Comment:
This isn't strictly needed since lmdb / leveldb data that is unencoded does not require OpenCV IO. I think `DataLayer` should only require lmdb / leveldb. If one tries to use encoded data without OpenCV the `DataTransformer` guards will report the dependency failure.